mr/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergObjectInspector.java

```diff
@@ -27,12 +27,25 @@
 import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
 import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
 import org.apache.iceberg.Schema;
+import org.apache.iceberg.common.DynMethods;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.TypeUtil;
 import org.apache.iceberg.types.Types;
 
 public final class IcebergObjectInspector extends TypeUtil.SchemaVisitor<ObjectInspector> {
 
+  // get the correct inspectors depending on whether we're working with Hive2 or Hive3 dependencies
+  // we need to do this because there is a breaking API change in Date/TimestampObjectInspector between Hive2 and Hive3
+  private static final DynMethods.StaticMethod DATE_INSPECTOR = DynMethods.builder("get")
+      .impl("org.apache.iceberg.mr.hive.serde.objectinspector.IcebergDateObjectInspectorHive3", null)
+      .impl(IcebergDateObjectInspector.class, null)
```

Comment:
It is safer to use the class name string instead of the class here, so that we know it won't be loaded and break at runtime.

Reply:
yeah, good idea to be defensive